dig.go

```diff
@@ -265,12 +256,61 @@ var _graphTmpl = template.Must(
 // Visualize parses the graph in Container c into DOT format and writes it to
 // io.Writer w.
 func Visualize(c *Container, w io.Writer, opts ...VisualizeOption) error {
-	if err := _graphTmpl.Execute(w, c.dg); err != nil {
+	dg := c.createGraph()
+
+	if err := _graphTmpl.Execute(w, dg); err != nil {
```

Comment:
nit: If there is enough context in the error message (without the additional formatting), we could write: 
```
func Visualize(c *Container, w io.Writer, opts ...VisualizeOption) error {
  return _graphTmpl.Execute(w, c.createGraph())
}
```

What do you think?

Reply:
That's neat. The error itself is indeed pretty self-explanatory and I removed error formatting.